internal/database/dbconn/dbconn_restricted.go

```diff
@@ -0,0 +1,82 @@
+package dbconn
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	// Restricted is the "sg_service" DB connection used to provide row level security.
+	// See https://docs.sourcegraph.com/admin/repo/row_level_security for more information.
+	// Only use this after a call to SetupRestrictedConnection.
+	Restricted *pgxpool.Pool
+)
+
+// SetupRestrictedConnection connects to the given data source, ensures reduced
+// privileges, and stores the handle globally.
+//
+// Each connection in the pool will adopt the "sg_service" role upon being checked
+// out. This role has row security policies applied to it so that various authn/z
+// checks are handled at the Postgres level instead of application code. This lowers
+// the burden on developers, since you only have to opt-in to using the connection
+// vs. invoking specific functions at specific times.
+//
+// The restricted connection is configured after the base connection, so we don't
+// need to handle the various liveness checks (e.g. is the database up). If we
+// reached this point, it is.
```

Comment:
Updating comment to reflect intent rather than guarantee. :grinning: 

```suggestion
// SetupRestrictedConnection does not handle the various liveness checks
// (e.g. is the database up). Thus the restricted connection must be configured after the
//  base connection is set up in SetupGlobalConnection.
```